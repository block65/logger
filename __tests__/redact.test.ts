import {
  afterEach,
  beforeEach,
  describe,
  expect,
  jest,
  test,
} from '@jest/globals';
import { createRedactProcessor } from '../lib/index.js';
import { createLoggerWithWaitableMock } from './helpers.js';

describe('Redact', () => {
  beforeEach(() => {
    jest.useFakeTimers('modern');
    jest.setSystemTime(new Date('2009-02-13T23:31:30.000Z'));
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  test('Basic', async () => {
    const [logger, callback, errback] = createLoggerWithWaitableMock({
      processors: [
        createRedactProcessor({
          censor: 'ðŸ§±ðŸ§±ðŸ§±ðŸ§±',
          paths: ['secret'],
        }),
      ],
    });

    logger.error({ secret: 'trustno1' });
    logger.error('trustno1');

    await logger.flush();

    await expect(callback.waitUntilCalledTimes(1)).resolves.toMatchSnapshot();
    expect(errback).not.toBeCalled();
  });
});
