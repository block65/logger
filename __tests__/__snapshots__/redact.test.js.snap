// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redact Basic 1`] = `
Array [
  Array [
    Object {
      "data": Object {
        "secret": "ðŸ§±ðŸ§±ðŸ§±ðŸ§±",
      },
      "level": 50,
      "msg": undefined,
      "time": 2009-02-13T23:31:30.000Z,
    },
  ],
  Array [
    Object {
      "level": 50,
      "msg": "trustno1",
      "time": 2009-02-13T23:31:30.000Z,
    },
  ],
]
`;
